core/shared/src/main/scala/matryoshka/Recursive.scala

```diff
@@ -101,13 +115,37 @@ trait Recursive[T] extends Based[T] {
       : A =
     gcata[(B, ?), A](t)(distZygo(f), g)
 
+  def zygoM[A, B, M[_]: Monad]
+    (t: T)
+    (f: AlgebraM[M, Base, B], g: GAlgebraM[(B, ?), M, Base, A])
+    (implicit BT: Traverse[Base])
+      : M[A] = {
+    val k = distZygoM(f, distApplicative[Base, M])
+
+    def loop[W[_]: Comonad : Traverse]
+      (t: T)
+      (g: GAlgebraM[W, M, Base, A], k: DistributiveLaw[Base, (M ∘ W)#λ])
```

Comment:
You can get rid of these parameters, because `loop` always uses the same `g` and `k`.